### PR TITLE
Fix get_area_list

### PR DESCRIPTION
**Pull request by tgrandje**
Should resolve https://github.com/InseeFrLab/pynsee/issues/145

### CODE REVIEW

**Comment by tgrandje** [comment]
Just added some more changes to get_area_list. To sum it up, the PR contains :

* a quick and dirty fix of area's case sensitive problem (which would happen with areas rightfully containing upper-case letters, for instance "collectivitesDOutreMer")
* make get_area_list consistent with get_geo_list about "date" argument : 
  * add the argument to the function 
  * change default behaviour (which until now used a "*" wildcard and so got all available data) to the same as get_geo_list : the default is now consistent with the API's documentation (ie. the current date)
* black formatting

File: pynsee/localdata/get_area_list.py [hunk 1]
```diff
@@ -49,40 +54,49 @@ def get_area_list(area=None, update=False):
...
             list_available_area = [area]
 
-    filename = _hash("".join(['get_area_list'] + list_available_area))
+    date_hash = ""
+    if date is not None:
+        date_hash = date
+
+    filename = _hash(
+        "".join(["get_area_list"] + list_available_area + [date_hash])
+    )
```
Comment (tfardet) [inline pynsee/localdata/get_area_list.py:85]:
```suggestion
    filename = _hash(
        "".join(["get_area_list"] + list_available_area + [date or ""])
    )

```

**Review by tfardet — CHANGES_REQUESTED** [review]
Thanks for the bugfix!
Could you also add a test for these cases?

@hadrilec I'm not familiar with this INSEE data but I'm a bit surprised by the hardcoded date, is this something that will never change of will we see later entries in the future that would make it worth creating an issue to think about how to properly handle this?
(not relevant for that PR, obviously)

**Comment by tgrandje** [comment]
@tfardet that one should be ok now, though I didn't checked the behaviour of the tests. Those are covered in the automated tests directly runned on github, right ?

File: pynsee/localdata/get_area_list.py [hunk 2]
```diff
@@ -110,15 +121,17 @@ def get_area_list(area=None, update=False):
...
     else:
         try:
             data_all = pd.read_pickle(file_data)
-        except:
+        except Exception:
             os.remove(file_data)
             data_all = get_area_list(area=area, update=True)
         else:
-            logger.info(
+            print(
```
Comment (tfardet) [inline pynsee/localdata/get_area_list.py:134]: something wrong happened here ^^

**Review by tfardet — CHANGES_REQUESTED** [review]
Almost there: you seem to have undone some of your previous hard work by reintegrating a print...

**Review by tfardet — APPROVED** [review]

**Comment by tgrandje** [comment]
Sorry, my bad (I should have postponed it to this morning :-) ). Github desktop displayed some strange things while changing of computer this morning but this seems ok on Github (web).

**Comment by hadrilec** [comment]
@tfardet 

> Thanks for the bugfix! Could you also pull from main branch and add a test for these cases?
> 
> @hadrilec I'm not familiar with this INSEE data but I'm a bit surprised by the hardcoded date, is this something that will never change of will we see later entries in the future that would make it worth creating an issue to think about how to properly handle this? (not relevant for that PR, obviously)

Indeed I agree, hard coded dates are far from being state-of-the-art. It is good to know that on paper local data API is still on a beta version. Then, some data update occurs every 5 five years so it is not a big deal to get some dates hard coded as long as the maintenance can keep up.

**Comment by hadrilec** [comment]
Thanks a lot @tgrandje and @tfardet. Just merged after review.

**Pull request merged**